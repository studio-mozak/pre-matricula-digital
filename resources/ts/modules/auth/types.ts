export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
