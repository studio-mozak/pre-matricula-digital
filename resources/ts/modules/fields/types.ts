import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
