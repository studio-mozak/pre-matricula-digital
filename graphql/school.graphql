extend type Query @guard(with: ["web", "sanctum", "prematricula"]) {

  """
  Retorna todas as escolas.
  """
  schools(
    """
    Retorna apenas escolas de IDs.
    """
    schools: [ID!] @in(key: "id")

    """
    Retorna apenas escolas que possuam turmas vínculadas a um processo de pré-matrícula.
    """
    processes: [ID!] @scope(name: "processes")
  ): [School!]! @paginate

  school(id: ID @eq): School @find

  getGroupedVacancies(
    school: ID!
    schoolYear: ID!
    grades: [ID!]!
    periods: [ID!]!
  ): [GroupedVacancies!]!

}

type School {
  id: ID!
  name: String!
  latitude: Float
  longitude: Float
  area_code: Int
  phone: String
  email: String
  processes: [Process!]! @belongsToMany
  classrooms(
    schoolYear: ID! @eq(key: "school_year_id")
    grades: [ID!] @in(key: "grade_id")
    periods: [ID!] @in(key: "period_id")
  ): [Classroom!]! @hasMany(type: PAGINATOR)
  vacancies: Int
}

type GroupedVacancies {
  grade: Grade!
  period: Period!
  school: School!
  vacancies: Int!
}
