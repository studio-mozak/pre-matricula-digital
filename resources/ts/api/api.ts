import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
