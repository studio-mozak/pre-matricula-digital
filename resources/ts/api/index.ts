export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
