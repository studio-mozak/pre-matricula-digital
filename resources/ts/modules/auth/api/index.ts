/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
