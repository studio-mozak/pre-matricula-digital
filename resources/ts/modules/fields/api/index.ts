/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
