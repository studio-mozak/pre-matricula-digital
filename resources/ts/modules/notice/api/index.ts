/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Notice from './services/graphql/notice';
