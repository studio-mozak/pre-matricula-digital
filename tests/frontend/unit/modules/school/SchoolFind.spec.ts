/// <reference types="@types/google.maps" />
import { Processes, ShowVacanciesReturn } from '@/modules/school/types';
import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import SchoolFind from '@/modules/school/components/SchoolFind.vue';
import { Schoolfind as SchoolFindApi } from '@/modules/school/api';
import { createTestingPinia } from '@pinia/testing';
import { initialize } from '../../mocks/google';
import { expect, vi } from 'vitest';

vi.mock('@/modules/school/api');

const procesesMock: Processes[] = [
  {
    id: '1',
    name: 'Processo de Testes 1',
    grades: [
      { id: '27', name: 'Nível I' },
      { id: '5', name: 'Nivel II' },
      { id: '44', name: 'Maternal' },
      { id: '12', name: 'Berçário' },
    ],
    schoolYear: {
      year: 2021,
    },
    stages: [
      {
        endAt: '2020-10-19 23:59:00',
        id: '1',
        name: 'Test',
        startAt: '2020-10-14 00:00:00',
        status: 'CLOSED',
        type: 'REGISTRATION_RENEWAL',
      },
      {
        endAt: '2020-10-27 23:59:00',
        id: '2',
        name: 'Test',
        startAt: '2020-10-22 00:00:00',
        status: 'CLOSED',
        type: 'REGISTRATION',
      },
      {
        endAt: '2020-12-18 17:00:00',
        id: '12',
        name: 'Test',
        startAt: '2020-10-23 15:00:00',
        status: 'CLOSED',
        type: 'WAITING_LIST',
      },
    ],
  },
];

const loaderVacanciesMock: ShowVacanciesReturn = {
  vacancies: [
    {
      grade: '5',
      period: '2',
      process: '1',
      school: '1',
    },
    {
      grade: '5',
      period: '4',
      process: '1',
      school: '1',
    },
    {
      grade: '12',
      period: '2',
      process: '1',
      school: '1',
    },
    {
      grade: '12',
      period: '4',
      process: '1',
      school: '1',
    },
    {
      grade: '27',
      period: '2',
      process: '1',
      school: '1',
    },
    {
      grade: '27',
      period: '4',
      process: '1',
      school: '1',
    },
    {
      grade: '44',
      period: '2',
      process: '1',
      school: '1',
    },
    {
      grade: '44',
      period: '4',
      process: '1',
      school: '1',
    },
    {
      grade: '12',
      period: '2',
      process: '1',
      school: '703011',
    },
    {
      grade: '12',
      period: '4',
      process: '1',
      school: '703011',
    },
  ],
  schools: [
    {
      area_code: 47,
      id: '0000001',
      lat: -26.257815,
      lng: -49.52137,
      name: 'Escola Municipal de Testes 1',
      phone: '9999-9999',
      position: {
        lat: -26.257815,
        lng: -49.52137,
      },
    },
    {
      area_code: 47,
      id: '0000002',
      lat: -26.264827,
      lng: -49.514835,
      name: 'Escola Municipal de Testes 2',
      phone: '9999-9999',
      position: {
        lat: -26.264827,
        lng: -49.514835,
      },
    },
    {
      area_code: 47,
      id: '0000003',
      lat: -26.24461,
      lng: -49.540354,
      name: 'Escola Municipal de Testes 3',
      phone: '9999-9999',
      position: {
        lat: -26.24461,
        lng: -49.540354,
      },
    },
    {
      area_code: 47,
      id: '0000004',
      lat: -26.227579,
      lng: -49.511398,
      name: 'Escola Municipal de Testes 4',
      phone: '9999-9999',
      position: {
        lat: -26.227579,
        lng: -49.511398,
      },
    },
    {
      area_code: 47,
      id: '0000005',
      lat: -26.269295,
      lng: -49.553098,
      name: 'Escola Municipal de Testes 5',
      phone: '9999-9999',
      position: {
        lat: -26.269295,
        lng: -49.553098,
      },
    },
  ],
};

const onMounted = async (
  success: boolean,
  proceses = procesesMock,
  loaderVacancies = loaderVacanciesMock
) => {
  if (success) {
    vi.mocked(SchoolFindApi.showProcesses).mockResolvedValueOnce(proceses);
    vi.mocked(SchoolFindApi.showVacancies).mockResolvedValueOnce(
      loaderVacancies
    );
  } else {
    vi.mocked(SchoolFindApi.showProcesses).mockRejectedValueOnce([]);
    vi.mocked(SchoolFindApi.showVacancies).mockRejectedValueOnce({});
  }

  const wrapper = mount(SchoolFind, {
    global: {
      plugins: [createTestingPinia()],
      stubs: [
        'x-field',
        'google-maps',
        'google-maps-marker-component',
        'google-maps-markers',
      ],
      mocks: {
        google: vi.fn(() => google),
      },
    },
  });

  await flushPromises();

  await wrapper.vm.$nextTick();

  return wrapper;
};

describe('SchoolFind', () => {
  let wrapper: VueWrapper<InstanceType<typeof SchoolFind>>;

  beforeEach(async () => {
    initialize();

    wrapper = await onMounted(true);

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('the initial variables must be loaded with the correct data', () => {
    expect(wrapper.vm.processes).toMatchObject(procesesMock);
    expect(wrapper.vm.vacancies).toMatchObject(loaderVacanciesMock.vacancies);
    expect(wrapper.vm.schools).toMatchObject(loaderVacanciesMock.schools);

    expect(wrapper.vm.schoolsInMap.length).toBe(5);
  });

  test('This should leave the variables empty, in case an error occurs in the request', async () => {
    wrapper = await onMounted(false);

    await wrapper.vm.$nextTick();

    await flushPromises();

    expect(wrapper.vm.processes).toMatchObject([]);
    expect(wrapper.vm.vacancies).toMatchObject([]);
    expect(wrapper.vm.schools).toMatchObject([]);

    expect(wrapper.vm.schoolsInMap.length).toBe(0);
  });

  test('should do a successful address search', async () => {
    const address = 'Rua Teste, 123';

    wrapper.vm.address = address;
    wrapper.vm.grade = '12';

    expect(wrapper.vm.address).toBe(address);
    expect(wrapper.vm.grade).toBe('12');

    const btn = wrapper.find('[data-test="btn-search-school"]');

    await btn.trigger('click');

    await wrapper.vm.$nextTick();

    await flushPromises();

    expect(wrapper.vm.schoolsInMap.length).toBe(1);
  });
});
