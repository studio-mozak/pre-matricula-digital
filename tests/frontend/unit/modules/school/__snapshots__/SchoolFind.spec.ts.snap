// Vitest Snapshot v1

exports[`SchoolFind > should match snapshot 1`] = `
<div>
  <div
    class="row mt-3"
  >
    <div
      class="container-sm col-10 col-lg-6 mt-4 pt-lg-4"
    >
      <div
        class="d-flex align-items-center mb-5"
      >
        <div
          class="font-muli"
          style="color: #003473;"
        >
          <div
            style="font-size: 24px;"
          >
            Some slogan
          </div>
          <div
            style="font-size: 34px;"
          >
            Some City
          </div>
        </div>
        <div
          class="ml-auto col-5"
        >
          <img
            alt="Some City"
            class="img-fluid"
            src="https://logo.com"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      <h2
        class="title-find-school"
      >
        Consultar escola
      </h2>
      <div
        class="x-card x-card--flat no-shadow bg-primary mt-4"
      >
        <div
          class="x-card__section x-card__section--vert"
        >
          <form>
            <x-field-stub
              allowclear="false"
              bind="[object Object]"
              containerclass="w-100"
              disabled="false"
              errors="false"
              label="Endereço"
              labelclass="form-label-bg-blue"
              loading="false"
              name="address"
              options=""
              placeholder="Digite o seu endereço (rua e número)"
              searchable="false"
              type="TEXT"
            />
            <div
              class="mt-2"
            >
              <div
                class="d-flex justify-content-between"
              >
                <x-field-stub
                  allowclear="false"
                  bind="[object Object]"
                  containerclass="mr-3 w-100"
                  disabled="false"
                  errors="false"
                  id="grade"
                  label="Série"
                  labelclass="form-label-bg-blue"
                  loading="false"
                  name="grade"
                  options="[object Object],[object Object],[object Object],[object Object]"
                  placeholder="Selecione a série"
                  searchable="true"
                  type="SELECT"
                />
                <div
                  class="d-flex align-items-center mt-3"
                >
                  <button
                    class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle x-btn--actionable pmd-focusable pmd-hoverable w-100 bg-primary-light text-primary flex-row"
                    data-test="btn-search-school"
                    role="button"
                    style="height: 45px;"
                    tabindex="0"
                    type="submit"
                  >
                    <span
                      class="x-btn__content text-center d-flex justify-content-center"
                    >
                      <i
                        aria-hidden="true"
                        class="fa fa-search pmd-icon text-"
                        role="img"
                        style="margin-top: 1px;"
                      >
                         
                      </i>
                    </span>
                    <transition-stub
                      appear="false"
                      css="true"
                      name="x-transition--fade"
                      persisted="false"
                    >
                      <!---->
                    </transition-stub>
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-12 mt-5"
    >
      <div
        class="card"
      >
        <google-maps-stub
          config="[object Object]"
          lat="0"
          lng="0"
          style="height: 400px;"
          zoom="13"
        />
      </div>
    </div>
  </div>
</div>
`;
